### PR TITLE
Added UI

### PR DESCRIPTION
**Pull request by theBwoyProgrammer**
## OOP School Library
- Create an `app.rb` file that will serves as my console app entry-point
- Added methods in my `app.rb` listing all books and people
- Created a person (teacher or student, not a plain Person)
- Created a `book`.
- Created a` rental.`
- Listed all rentals for a given `person id.`
- Defined the entry point in `main.rb`
- Presented the user with a list of options to perform.
- Let users choose an option
- Asked for parameters for the option
- Added a way to quit the app

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  def list_people
+    people = [*@teacher, *@student]
+    people.each_with_index { |person, i| print "(#{i}) Name: \"#{person.name}\", Age: \"#{person.age}\"\n" }
+  end
+
+  def create_person # rubocop:disable Metrics/MethodLength
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    input = gets.chomp.to_i
+    case input
+    when 1
```
Comment (GzimAsani) [inline app.rb:30]: - [x] Add a checker here, if the user inputs something other than 1 or 2 example: 
```ruby
    if person_type != "1" && person_type != "2"
      puts "Invalid option"
      return
    end
```

Comment (GzimAsani) [reply app.rb:30]: - [x] If the user selects 1 which means he is a student, kindly ask the user if he has parent permission
`print "Has parent permission? [Y/N]: "`

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+require_relative 'App'
```
Comment (GzimAsani) [inline main.rb]: - [x] This should be lowercase `app`

File: app.rb (hunk 1 above)
Comment (GzimAsani) [reply app.rb:30]: - [x] Kindly check this error, I am unable to create a new user. I suspect the error comes from the student class
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @techwolf404         :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your project, it is almost done:+1:  
 
 ## Required Changes 

- [ ] Check the comment below :arrow_double_down: 






 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (theBwoyProgrammer) [reply app.rb:30]: I have fixed them

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
```
Comment (tresorsawasawa) [inline app.rb:30]: - [x] As mentioned by the previous reviewer and as required for the project, kindly ensure your application works as expected. As pointed out in the last review, right now users cannot create a new `Person` for both `teacher` and `student`. Below are the error images, I am getting when trying so:

| Teacher Error | Student |
| ------------ | -------------- |
| Screenshot from 2022-10-06 21-02-46 | Screenshot from 2022-10-06 21-10-15 |

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUEST CHANGES :recycle::recycle: 

Hi @techwolf404 
Great work on making the changes requested by the previous reviewer  👏👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 3 above)
Comment (theBwoyProgrammer) [reply app.rb:30]: I hadn't realized that. But I have implemented them

**Review by julie-ify — APPROVED** [review]
Hi @techwolf404,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**